code/modules/hivemind/mobs.dm

```diff
@@ -0,0 +1,916 @@
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	qdel(src)
+
+///we live to/////////////////////////////////////////////////////////////////
+////////////////////////////////////SMALL GUYS///////////////////////////////
+//////////////////////////////////////////////////////////////die for hive//
+
+
+/////////////////////////////////////STINGER//////////////////////////////////
+//Special ability: none
+//Just another boring mob without any cool abilities
+//High chance of malfunction
+//Default speaking chance
+//Appears from dead small mobs or from hive spawner
+//////////////////////////////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/hivemind/stinger
+	name = "medbot"
+	desc = "Ah-h little good guy. Wait a minute, is this a blade? Yep. You see a whole blade in place where syringe must be."
+	icon_state = "slicer"
+	attacktext = "moves around and slice"
+	density = 0
+	speak_chance = 3
+	malfunction_chance = 15
+	mob_size = MOB_SMALL
+	pass_flags = PASSTABLE
+	speed = 4
+
+	speak = list(
+				"I've seen this ai. Ma-an, that's aw-we-e-ewful!",
+				"I know, i know, i remember this one.",
+				"Rad-d-dar, put a ma-ma... mask on!",
+				"Delicious! Delicious... Del-delicious?..",
+				)
+	target_speak = list(
+				"Hey, i'm comming!",
+				"Hold on! I'm almost there!",
+				"I'll help you! Come closer.",
+				"Only one healthy prick!",
+				"He-e-ey?"
+				)
+
+
+/mob/living/simple_animal/hostile/hivemind/stinger/death()
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	qdel(src)
+
+
+/////////////////////////////////////BOMBER///////////////////////////////////
+//Special ability: none
+//Explode in contact with target
+//High chance of malfunction
+//Default speaking chance
+//Appears from dead small mobs or from hive spawner
+//////////////////////////////////////////////////////////////////////////////
+
+/mob/living/simple_animal/hostile/hivemind/bomber
+	name = "bot"
+	desc = "This one looks fine. Only sometimes it's careen from one side to another."
+	icon_state = "bomber"
+	density = 0
+	speak_chance = 3
+	malfunction_chance = 15
+	mob_size = MOB_SMALL
+	pass_flags = PASSTABLE
+	speed = 6
+
+	speak = list(
+				"Can you help me, please? There's something strange.",
+				"Are you... Are you kidding?",
+				"I want to pass away, just trying to get out from here.",
+				"A really bad place. We are in deep shit here.",
+				"I'm not sure that we just can stop it.",
+				)
+	target_speak = list(
+						"Here you are! I have something for you. Something special!",
+						"Hey! Hey? Help me, please!",
+						"Hey, look, look. I'll not harm you, just calm down!",
+						"Oh god, this is... Yes, this is what we are looking for."
+						)
+
+	New()
+		..()
+		set_light(2, 1, "#820D1C")
+
+
+/mob/living/simple_animal/hostile/hivemind/bomber/death()
+	..()
+	gibs(loc, null, /obj/effect/gibspawner/robot)
+	explosion(src.loc, 0, 0, 2)
```

Comment:
loc

Reply:
Actually, to prevent any wierdness, this should generally use get_turf(src)